Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -547,6 +560,52 @@ extension StripeCardReaderService {
         }
     }
 
+    /// Implements refund retry logic as recommended by Stripe
+    /// https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPTerminal.html#/c:objc(cs)SCPTerminal(im)processRefund
+    /// > When `processRefund` fails, the SDK returns an error that either includes the failed `SCPRefund` or the `SCPRefundParameters` that led to a failure.
+    /// > Your app should inspect the `SCPProcessRefundError` to decide how to proceed.
+    /// >
+    /// > If the `refund` property is nil, the request to Stripe’s servers timed out and the refund’s status is unknown.
+    /// > We recommend that you retry processRefund with the original `SCPRefundParameters`.
+    /// >
+    /// > If the `SCPProcessRefundError` has a `failure_reason`, the refund was declined.
+    /// > We recommend that you take action based on the decline code you received.
+    private func shouldRetryRefund(after processError: ProcessRefundError) -> Bool {
+        if let refund = processError.refund {
+            // Retry based on `failure_reason`
+            return shouldRetryRefund(with: refund.failureReason)
+        } else {
+            // Retry because the status is unknown
+            return true
+        }
+    }
+
+    private func shouldRetryRefund(with failureReason: String?) -> Bool {
+        guard let failureReason = failureReason else {
+            return false
+        }
+        switch DeclineReason(with: failureReason) {
+        case .fraud,
+                .invalidAccount,
+                .currencyNotSupported,
+                .cardNotSupported,
+                .duplicateTransaction,
+                .expiredCard,
```

Comment:
this decline reason is considered as retryable [in Android](https://github.com/woocommerce/woocommerce-android/blob/567742faae9786e774f419e02f48d171bdaba97c/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/payment/CardReaderPaymentViewState.kt#L138) (also confirmed again with Android in p1650533500277319/1650028971.895919-slack-C025A8VV728), do we want to follow? I think it makes sense for this to be retryable, because the customer can try using a different card (this happened to me once when I used to still keep the expired card 😅 )

Reply:
Ah yes, that's true, I moved that case to the "retryable" ones